rfc-216-use-static-typing-where-available.md

```diff
@@ -39,14 +39,32 @@ code will be expected to handle.
 Where a language either includes support for static typing or where tooling
 exists to ensure it, new code we write SHOULD make use of static typing.
 
-Where static typing is used, any automated test pipeline MUST make sure that
+Where static typing is used in an interpreted language, any automated test pipeline MUST make sure that
 types are checked.
 
 If an existing project is in a language which supports optional or partial
 typing, you MAY choose to add type annotations to some, all or none of the
 existing codebase. You SHOULD add type annotations to new and refactored code.
 
+### Languages
+
+The following tooling SHOULD be used in the following languages:
+
+| Language   | Tooling                          |
+| ---------- | -------------------------------- |
+| C#         | N/A                              |
+| Kotlin     | N/A                              |
+| PHP        | N/A                              |
```

Comment:
@RobjS @matthewpassmore do you have any thoughts on this? I think we probably haven't always been consistent about how we use annotations?

Reply:
I left PHP as N/A here because (unlike Ruby/Python/JS) the typing is actually integral to the language rather than needing you to run an extra set of tools to make it work. If there's a handy PSR for it then it should probably be included in the same vein as PEP 484 though!

Reply:
We've used [Psalm](https://psalm.dev/) for static analysis on some projects. It does more than enforcing typing, but that's a big part of it, and we may be able to configure it to only worry about typing.

I've tried using static typing in PHP before and found it a bit of a mixed bag, as the tools don't always exist within the language to do it properly (although that has improved with PHP8), which means you end up with a combination of types declared in language and types declared in docblocks, which can feel a bit clunky.

Reply:
mypy for Python isn't strictly speaking only for typing (although that's certainly its main purpose), and it can surface some other "this is obviously not right" stuff so I don't think recommending Psalm is necessarily a bad idea.

I think my preference would be for in-language type declarations, but that's only really because that's the way PEP-8 does it and consistency is good. No strong feelings if people actually working with PHP every day feel something else is better.

Reply:
We're no longer recommending specific typing tools -- I believe we can resolve this conversation.